Allow Mapbox to load local json map style definition

lbaey opened the pull request:
Hi
I needed to load a different than "native mapbox style" in my map.
I added the ability to load a json map style file definition with a new special url pointing at the "assets" nativescript directory.
The Url is `assets://` and points to the `/app/assets/` directory of your project.
Usage example : 
```
onMapReady(args) {
                console.log("Switching Style");
                args.map.setMapStyle("assets://RM-raster-style.json");
}
```
Tell me what you think and if you think this is a good approach.
Cheers

lbaey-mobizel committed:
Add http and file json style loading

lbaey-mobizel committed:
Merge test lines

lbaey-mobizel committed:
Use asset:// for android, as file:// is not supported

EddyVerbruggen commented:
Hi! I think that's a great approach, but perhaps it's worthwhile to explore the possibility of using `~/yourfile.json` instead because that's the standard NativeScript approach to reference files in the `/app` folder (`~` being and alias for `app`). Do you think that would be possible? That way users aren't forced to create an `assets` folder to store the file in.

Can you perhaps also add a sample json file to the demo in the repo? I'd love to play with it!

lbaey-mobizel committed:
Swith local URL path to nativescript local app definition ~/

lbaey commented:
OK, I was not sure on how to refer to local path, but the nativescript "~" seems ideal.
I've switched to this URL scheme tracking to let mapbox load the json file.
I've also added some doc and (untested and commented) example so you can play ;)

lbaey-mobizel committed:
Typo

EddyVerbruggen commented:
Awesome! I'll be testing this soon.

EddyVerbruggen commented:
Works like a charm! I only tested with `~/` though. THANKS!

ansarikhurshid786 commented:
I want load local json. can you give me an example which work in android and ios. 
Thanks in advance
